### PR TITLE
uplift/bot: Cancel uplift requests when merging fails.

### PR DESCRIPTION
**Pull request by jankeromnes**
When a Firefox patch is requested be uplifted, but it doesn't apply cleanly, we:
1. Remove the "approval-mozilla-{beta,release,esr*}" flag from attachments
2. Post a comment explaining that the merge failed, with details
3. Needinfo the person who requested the uplift

TODO:

- [x] Use a custom Bugzilla token to update bugs as [Uplift Bot [:upliftbot]](https://bugzilla.mozilla.org/user_profile?login=upliftbot%40mozilla.com)
- [x] Add a setting to disable bug editing

### CODE REVIEW

**Comment by jankeromnes** [comment]
@La0 Please take a look when you have time.

Regarding "Use a custom Bugzilla token", I notice that the `BUGZILLA_TOKEN` secret used in upliftbot comes from the common configuration section (under `common:` in the TaskCluster secret). If I simply add a new `BUGZILLA_TOKEN` entry to the `uplift/bot:` section, will it successfully overload the token used throughout upliftbot?

Regarding "Add a setting to disable bug editing", what would be the best way to achieve this in upliftbot? (I notice that there is no `settings` util, but `uplift_bot/cli.py` uses the `get_secrets` method in `main`, so maybe I can simply add another secret under `uplift/bot:` (e.g. called `EDIT_BUGZILLA: False`)?

File: src/uplift/bot/uplift_bot/bugzilla.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from libmozdata.bugzilla import Bugzilla
```
Comment (La0) [inline src/uplift/bot/uplift_bot/bugzilla.py]: Don't you need the latest version of libmozdata ? Uplift bot uses 0.1.40 right now

File: src/uplift/bot/uplift_bot/bugzilla.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        ],
+    }
+
+    def _attachmenthandler(attachments, bugid, needinfos):
+        for attachment in attachments:
+            flags = [f for f in attachment['flags'] if f['name'] == flag_name]
+            if len(flags) > 0:
+                # Remember who requested this uplift, in order to needinfo them
+                # later.
+                needinfos.add(flags[0]['setter'])
```
Comment (La0) [inline src/uplift/bot/uplift_bot/bugzilla.py]: This part should be in the backend, adding more infos to the bug api output

**Comment by jankeromnes** [comment]
> Yes you can use a custom overriding bugzilla token in the current project section.

Thanks! I'll do that then.

> And yes that's the way to go: add a secret and read it in main. Do you want to do this in this PR ?

Yes, I'll do it here, in order to prevent Bugzilla to be edited until we're confident about the results.

File: src/uplift/bot/uplift_bot/bugzilla.py (hunk 1 above)
Comment (jankeromnes) [reply src/uplift/bot/uplift_bot/bugzilla.py]: I do. Can we upgrade the version Uplift bot uses?

File: src/uplift/bot/uplift_bot/bugzilla.py (hunk 2 above)
Comment (jankeromnes) [reply src/uplift/bot/uplift_bot/bugzilla.py]: Which "backend" do you mean? (I thought that the backend part of uplift bot was to be refactored/removed soon.)

File: src/uplift/bot/uplift_bot/bugzilla.py (hunk 1 above)
Comment (jankeromnes) [reply src/uplift/bot/uplift_bot/bugzilla.py]: Awesome, thanks!

**Comment by jankeromnes** [comment]
I've added a `BUGZILLA_READ_ONLY` setting to be able to disable bug editing.

@La0 Please have another look when you have time.

**Comment by jankeromnes** [comment]
Rebased on top of master, but still blocked on #1497.

**Comment by La0** [comment]
@jankeromnes #1497 is now merged, you can rebase again and get all the fixes

**Comment by jankeromnes** [comment]
Rebased again.

**Comment by jankeromnes** [comment]
Fixed up after https://bugzilla.mozilla.org/show_bug.cgi?id=1486410#c26 and rebased.

**Pull request merged**